### PR TITLE
`tjson`: Cover `document` (`object`) type with tests

### LINKED ISSUE #912: `tjson`: Cover `document` (`object`) type with tests
This task is a part of #683 epic.
This task should be taken after #899 is done.

 🎯 The goal of it is to cover dealing with objects (`tjson`'s type `document`). 

The type itself is implemented but is not covered with tests. We can reuse test cases from `fjson`.

If needed, Tigris' documentation for types can be found [here](https://docs.tigrisdata.com/overview/schema).

To Do:
- [x] Add tests (`test`, `fuzz`, `benchmark`) to cover `document` type in the `tjson` package.

### PR DESCRIPTION
**Pull request by rumyantseva**
# Description

This PR closes #912.

## Readiness checklist

* [x] I added tests for new functionality or bugfixes.
* [x] I ran `task all`, and it passed.
* [x] I added/updated comments for both exported and unexported top-level declarations (functions, types, etc).
* [ ] I checked comments rendering with `task godocs`.
* [x] (for maintainers only) I set Reviewers, Assignee, Labels, Project and project's Sprint fields.
* [x] I marked all done items in this checklist.

### CODE REVIEW

File: internal/tjson/schema.go [hunk 1]
```diff
@@ -240,11 +241,34 @@ func DocumentSchema(doc *types.Document) (*Schema, error) {
 	return &schema, nil
 }
 
+// subdocumentSchema returns a JSON Schema for the given subdocument.
+// Subdocument is a "nested" document that can be used as a property of another document or subdocument.
```
Comment (rumyantseva) [inline internal/tjson/schema.go:245]: Let me know if you have a better idea for this case.

**Review by noisersup — DISMISSED** [review]
LGTM 

Thank you for descriptive comments they are helpful :)

File: internal/tjson/schema.go (hunk 1 above)
Comment (AlekSi) [reply internal/tjson/schema.go:245]: LGTM

File: internal/tjson/schema.go [hunk 2]
```diff
@@ -240,11 +241,35 @@ func DocumentSchema(doc *types.Document) (*Schema, error) {
 	return &schema, nil
```
[the remark this answers is not in the document]
Comment (rumyantseva) [reply internal/tjson/schema.go]: Fixed. PTAL.

**Review by AlekSi — APPROVED** [review]

**Review by w84thesun — APPROVED** [review]

**Pull request merged**